src/hooks/useZarrMetadata.ts

```diff
@@ -77,70 +145,43 @@ export default function useZarrMetadata() {
           fileBrowserState.currentFileSharePath.name,
           fileBrowserState.currentFileOrFolder.path
         );
-        const zarrayFile = fileBrowserState.files.find(
-          file => file.name === '.zarray'
-        );
+
+        const zarrayFile = await getFile('.zarray');
         if (zarrayFile) {
```

Comment:
It looks like we iterate through `fileBrowserState.files` a few times to look for three files. Perhaps it would make sense to accumulate these into a [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) once. We can then check the `Set` for the files via `has`.

The exact implementation is not specified other than the time complexity being better than `O(n)`, which is where we currently are.